notebooks/work-in-progress/fuel-price-imputation/fuel-price-imputation-zane.ipynb

```diff
@@ -0,0 +1,1133 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #3.    gkf = GroupKFold(n_splits=5)
</pre><p>Two things here:</p><ol><li>For the sake of reproducibility, I think we should set <code>random_state=&lt;your favorite number&gt;</code> for every piece that takes that arg. That includes this CV splitter, HistGBR, and maybe GridSearch too.</li><li>KFold splits data in the order it comes in (GroupKFold by the order the groups appear). Sometimes that is a useful thing, I'm not sure if it is here. I <em>think</em> frc comes roughly sorted by <code>report_date</code>, so this might be holding out plants roughly based on their age group. That sounds good for forecasting, but for imputation it might just add noise? I'm not totally sure. Why did you move from GroupShuffleSplit to GroupKFold?</li></ol><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/catalyst-cooperative/pudl/pull/1696/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Reply:
<p>I didn't really understand the difference between the various splitters, beyond ensuring that groups are partitioned, and that unevenly sized categories get proportionally represented in the different splits. It seems like there's no reason not to shuffle all the records within the group partitions that we want for test/train splits.</p>